Feature: Initial project setup and configuration (#1)
behrouzRajaei/v-server-setup: README.md

# v-server-setup

## Project Structur

v-server-setup
        ├── Checkliste.pdf

>         └── README.md

Reviewer: @behrouzRajaei you can use Code Blocks to show this part in better format. See GitHub Docs how to use code blocks: https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks

v-server-setup

## Project Structure
```
v-server-setup
├── Checkliste.pdf
└── README.md
```
---

## Technologies Used

* OpenSSH (key-based authentication only)
* NGINX Web Server
* Git + GitHub

---

## Prerequisites

* Linux Server
* GitHub account

---

## Server Setup Overview

### 1. SSH Access Configuration

* Created a new SSH key pair

```bash
ssh-keygen -t ed25519 -C "<username>@<ip_address>"
```

* Uploaded public key to the server

```bash
ssh-copy-id <username>@<ip_address>
```

* Disabled password login and PAM

```bash
sudo nano /etc/ssh/sshd_config
```

* Change the folowing items in sshd_config file
```
├── PasswordAuthentication no

> ├── UsePAM no
> └── PermitRootlogin no

Reviewer: @behrouzRajaei can your argument why do you change these settings? If you do not have a particular reason, please remove them.
```suggestion

```

```
* Restart service SSH

```bash
sudo sshd -t
sudo systemctl restart ssh
```

### 2. NGINX Web Server

> * Installed and verified NGINX is up and running

Reviewer: ```suggestion
* Installe and verifie NGINX is up and running

```

```bash
sudo apt update
sudo apt install nginx -y
sudo systemctl start nginx
sudo systemctl enable nginx
```

* Create alternative for NGINX seite

```bash
cd /var/www/html
sudo nano index.html
```

* Edit the default `index.html` file and overwrite it with your own title and content to replace the default Nginx welcome page.
* Test the website in browser
http://<ip_address>

### 4. Git + GitHub Integration

* Installed and verified Git

```bash
sudo apt install git
```

* Configure git by username and email_address

```bash
git config --global user.name "<your_username>"
git config --global user.email "<your_email@example.com>"
```
* Clone the project repository using SSH:

Go to your GitHub repository, click the green Code button, and select the SSH tab. Copy the SSH URL, then run:

```bash
git clone <your-ssh-url-here>

> ```

Reviewer: @behrouzRajaei before cloning, you should document how to generate a ssh-key-pair and integrate it with GitHub. https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent

> * Project version-controlled and documented.

Reviewer: @behrouzRajaei what does this mean?

## How to Use

* Clone the project to your local machine:
```bash
git clone <your-ssh-url-here>
cd v-server-setup
```

> * The project is now version-controlled and connected to GitHub.

Reviewer: @behrouzRajaei this section is already present in the section above. We do not want to repeat ourselves.

### 1. SSH Access Configuration

> * Created a new SSH key pair

Reviewer: @behrouzRajaei try not to write on the past. 
```suggestion
* Create a new SSH key pair

```

```bash

>         ssh-keygen -t ed25519 -C "brajaei@188.245.220.182"

Reviewer: @behrouzRajaei replace your sensitive Data with placeholders.

```suggestion
        ssh-keygen -t ed25519 -C "<username>@<ip_address>"

```

```

> * Uploaded public key to the server

Reviewer: ```suggestion
* Uploade public key to the server

```

```bash

>         ssh-copy-id brajaei@188.245.220.182

Reviewer: @behrouzRajaei replace your sensitive Data with placeholders.
```suggestion
        ssh-copy-id <username>@<ip_address>
```

```

> * Disabled password login and PAM

Reviewer: ```suggestion
* Disable password login and PAM

```

```bash
        sudo nano /etc/ssh/sshd_config
```

* Change the folowing items in sshd_config file
        ├── PasswordAuthentication no
        ├── UsePAM no

>         └── PermitRootlogin no

Reviewer: @behrouzRajaei use code blocks for better formatting

````suggestion
```bash
        ├── PasswordAuthentication no
        ├── UsePAM no
        └── PermitRootlogin no
```
````

* Restart service SSH

```bash
        sudo sshd -t
        sudo systemctl restart ssh
```

### 2. NGINX Web Server

> * Installed and verified NGINX is up and running

Reviewer: ```suggestion
* Installe and verifie NGINX is up and running

```

```bash
        sudo apt update
        sudo apt install nginx -y
        sudo systemctl start nginx
        sudo systemctl enable nginx
```

* Create alternative for NGINX seite

```bash
        cd /var/www/html
        sudo nano index.html
```

* Write the new Title and Content for new webseite
* Test the website in browser

>         http://188.245.220.182

Reviewer: ```suggestion
       http://ip_address
```

### 4. Git + GitHub Integration

* Installed and verified Git

```bash
        sudo apt install git
```

> * Configure git by User_name and Email Address

Reviewer: ```suggestion
* Configure git username and email

```

```bash
        git config --global user.name "Behrouz Rajaei"
        git config --global user.email "bmd.rajaei@gmail.com"
```

* Stage changes and commit them to the development branch

```bash
        git add .
        git commit -m "Projekt_data in devel;pment added"
```

* Push branch to GitHub

```bash
        git push --set-upstream origin development
```

> * Project version-controlled and documented.

Reviewer: @behrouzRajaei replace this with setting ssh connection with GitHub.
See the documentation on GitHub on how to do it: https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account

---

## Prerequisites

* Linux Server

> * GitHub account

Reviewer: @behrouzRajaei this section should be placed on top, before you start explaining the setup.

---

## How to Use

* Git initialized and connected to [GitHub repo](https://github.com/behrouzRajaei/v-server-setup) and Clone the project to your local machine:

```bash
        git clone git@github.com:behrouzRajaei/v-server-setup.git
        cd v-server-setup

> ```

Reviewer: @behrouzRajaei you can replace this with a command to test git ssh connection to GitHub. See documentation on how to test: https://docs.github.com/en/authentication/connecting-to-github-with-ssh/testing-your-ssh-connection
